### ISSUE TITLE
Pyfa crashes when deleting a saved fit

### ISSUE THREAD

**Issue by CountVonn** [pyfa-org/Pyfa#2023]
## Bug Report

After upgrading to pyfa 2.10.0 on OSX 10.13.6, deleting any saved fit causes pyfa to crash.  This occurs both for fits created before upgrading to 2.10.0, and for new fits created after upgrading to 2.10.0.  The contents of the fit do not seem to affect the behavior in any way, an empty fit manually created after the upgrade is crashing pyfa the same as fully populated fits, both created before or imported after the upgrade.

### Expected behavior:

Pyfa deletes the fit without crashing.

### Actual behavior:

Pyfa crashes after deleting the fit.  After reloading the program the fit has been deleted.

### Detailed steps to reproduce:

Create a new fit and save it, or import a new fit, or find an existing fit using the search tool.  Delete the fit and Pyfa crashes.  The occurs whether or not the fit is currently displayed in a tab.

### Fits involved in EFT format (Edit > To Clipboard > EFT):

Deleting this empty fit triggered a crash:

[Astero, Astero fit]

[Empty Low slot]
[Empty Low slot]
[Empty Low slot]
[Empty Low slot]

[Empty Med slot]
[Empty Med slot]
[Empty Med slot]
[Empty Med slot]

[Empty High slot]
[Empty High slot]

[Empty Rig slot]
[Empty Rig slot]
[Empty Rig slot]

### Release or development git branch? Please note the release version or commit hash:

Pyfa release 2.10.0

### Operating system and version (eg: Windows 10, OS X 10.9, OS X 10.11, Ubuntu 16.10):

OSX 10.13.6

### Other relevant information:

Here is the crash report generated by OSX:
[Pyfa crash report.txt](https://github.com/pyfa-org/Pyfa/files/3531109/Pyfa.crash.report.txt)

**Comment by DarkFenX** [comment]
Does it always crash, or randomly?

I will look into this issue but only next week - won't have access to Mac before that.

**Comment by CountVonn** [comment]
@DarkFenX In my testing it's a consistent crash.  On my MacBook, pyfa crashes every time a fit is deleted.

**Comment by DarkFenX** [comment]
What a shame. This time i actually more or less thoroughly tested release on Mac OS, but never got to deleting a fit.

**Comment by DarkFenX** [comment]
Change-wise, I see very few changes which could cause that. Maybe only upgrade of wxPython from 4.0.4 to 4.0.6. Unfortunately, trasfer.sh already removed builds right before and after this change, so I will have to make new build to verify this claim.

**Comment by DarkFenX** [comment]
Reproduced the issue on my mac as well. Following commit causes regression:
```
a028ebe198be75e9b9bbc050c769741c32809f8b is the first bad commit
commit a028ebe198be75e9b9bbc050c769741c32809f8b
Author: DarkPhoenix <phoenix@mail.ru>
Date:   Sat Aug 10 01:56:43 2019 +0300

    Rework how we handle all modal dialogs

:040000 040000 999b8aca579ffc43209ef8dff4ed2800dc8e8f17 e11651ed14e5c928a8a0af5dc7f6cb02c28b1d82 M	gui
:040000 040000 301085d3493bca2a6336cffaffb85bf8077feb5b 9fe0ec93edb8eb80dbe203bfe647a9359333ccc8 M	tests
```

As this issue is caused by modal dialog handling - you can work around it by deleting fit without dialog (clicking delete button while holding shift).

**Comment by DarkFenX** [comment]
Looks like following happens:

- dialog is opened with context manager in FitItem object
- when dialog is closed,we request pyfa to delete the fit which emits stage 3 ship browser message
- when processing this message, ship browser destroys all FitItems seen in current stage to refresh
- when done with stage 3, flow returns to FitItem object's method and python exits dialog context manager, which tries to destroy it
- as parent FitItem has been destroyed already, something weird happens and wx crashes (no idea why only on wx)

Fix would be to delay fit deletion until after dialog has been destroyed

**Comment by DarkFenX** [comment]
Filled an upstream issue https://github.com/wxWidgets/Phoenix/issues/1343

**Comment by DarkFenX** [comment]
Fixed in 5a34db0d2f5670462076b3df76c9766c57990c61 by not using dialog context manager which fucks up dialog destruction. After there's more info in linked ticket - will fix it properly.
